src/engine/client/ClientApplication.cpp

```diff
@@ -95,7 +100,7 @@ class ClientApplication : public Application {
         void Shutdown(bool error, Str::StringRef message) override {
             #if defined(_WIN32) || defined(BUILD_GRAPHICAL_CLIENT)
```

Comment:
Make this ifdef have 3 branches and put everything here instead of having a macro and stuff at the top of the file.

Reply:
I don't think this is possible. For a start `extern SDL_Window *window;` must be defined only if `BUILD_GRAPHICAL_CLIENT` (the type doesn't exist otherwise), and must be defined out of the function because C++.

Reply:
Oh I forgot it's inside a namespace. Maybe it would be better to have a ShowError helper function at the top... I'll stop insisting on any specific thing though. LGTM